src/ripple/net/impl/RPCCall.cpp

```diff
@@ -1267,9 +1267,10 @@ struct RPCCallImp
     // Build the request.
     static void onRequest (std::string const& strMethod,
         Json::Value const& jvParams,
-        const std::unordered_map<std::string,
-        std::string>& mHeaders, std::string const& strPath,
-        boost::asio::streambuf& sb, std::string const& strHost,
+        std::unordered_map<std::string, std::string> const& mHeaders,
+        std::string const& strPath,
```

Comment:
Consider `std::string_view` instead of `std::string const&`

Reply:
`string_view` is very nice. Unfortunately `std::string_view` is C++-17 only and we're still on 14.

Reply:
Oh? I'm surprised...C++14 is a minor update, most people either stay on C++11 or go straight to C++17. What is the obstacle?

Reply:
@vinniefalco Mostly it's just concervatism and making sure we have a good solution for all our supported platforms (centos is usually the most behind). I expect we'll upgrade to 17 soon; there are good things in there that would make rippled nicer.

Reply:
You can always write
```
namespace ripple {
using string_view = boost::string_view;
} // ripple
```

And then change that to `std::string_view` later

Reply:
Is there some mechanism to ensure that the original ```char *``` for the string_view stays in scope? I understand that avoiding a string copy is desirable, but is going back to consciously managing the lifetime of strings worth it?

Reply:
Howard could give you a concise answer but I can put it a different way - are you okay making unnecessary copies of the string depending on the type of argument at the call site?

Reply:
> Consider `std::string_view` instead of `std::string const&`

Based on your suggestion, I put string_view in for the server-side part of handling headers, but not the part that you suggested because the client side of this is just for testing, and string_view lacks a hash comparitor for it to live in an unordered_map. Not worth the hassle just for a couple of unit tests. But for the server side handling of headers, I think it should be beneficial.

Reply:
Entirely reasonable